src/item.h

```diff
@@ -111,7 +111,13 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
         item &operator=( const item & ) = default;
         virtual ~item() = default;
 
-        item( const std::string new_type, int turn, bool rand = true );
```

Comment:
> Are you sure you got all invocations of this constructor?

Yes. I deleted the `rand` parameter then by hand went through every invocation until the code compiled without it.